### ISSUE TITLE
WebFont global existing causes Google Charts with custom fontName to throw exception `Cannot read properties of null (reading '__eventTarget')`

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#5572]
## Bug Description

We've had 3 reports of the following error after updating to the latest version of the plugin, 1.79.0
`Cannot read properties of null (reading '__eventTarget')`

This appears as a dashboard error, with more details below:

Cannot read properties of null (reading '__eventTarget')

```
    in GoogleChart
    in div
    in div
    in UserDimensionsPieChart
    in div
    in Cell
    in div
    in ForwardRef
    in div
    in ForwardRef
    in div
    in div
    in Widget
    in WithWidgetSlug(Widget)
    in DashboardAllTrafficWidget
    in WhenAnalyticsActive(DashboardAllTrafficWidget)
    in WidgetRenderer
    in div
    in Cell
    in WidgetCellWrapper
    in div
    in ForwardRef
    in div
    in div
    in ForwardRef
    in WidgetAreaRenderer
    in div
    in WidgetContextRenderer
    in DashboardMainApp
    in DashboardEntryPoint
    in RestoreSnapshots
    in ErrorHandler
    in StrictMode
    in Root
```

<details><summary>Console Errors</summary>

```
TypeError: Cannot read properties of null (reading ‘__eventTarget’)
at gvjs_Zm (jsapi_compiled_default_module.js:412:29)
at Object.gvjs_Ym [as addListener] (jsapi_compiled_default_module.js:411:392)
at googlesitekit-modules-search-console-227f86b858fbc87c820e.js:31:11241
at Ui (googlesitekit-vendor-4167c42429c777851281.js:7:724946)
at va (googlesitekit-vendor-4167c42429c777851281.js:7:747869)
at t.unstable_runWithPriority (googlesitekit-vendor-4167c42429c777851281.js:7:767492)
at Mr (googlesitekit-vendor-4167c42429c777851281.js:7:692598)
at Yj (googlesitekit-vendor-4167c42429c777851281.js:7:744298)
at Lj (googlesitekit-vendor-4167c42429c777851281.js:7:734658)
at googlesitekit-vendor-4167c42429c777851281.js:7:692889
```

</details>

## Steps to reproduce

The below are the only known steps to consistently recreate so far:
1. Set up a new WP site, with SK set up and Google Analytics also connected
2. Install and activate [the Futurio Extra plugin](https://wordpress.org/plugins/futurio-extra/)
3. Install and activate [the Futurio theme ](https://pcm.wordpress.org/themes/futurio/)
4. Install one of the theme demos (Customizer > One Click Demo Import)
5. After importing, revisit your SK dashboard. The error appears. If the error doesn't appear, change the reporting period, or wait until cached response expire. 

**Note that upon refreshing the page with the error the dashboards loaded as expected in my case, as the reporting period switches back to the 28 day report, with the dashboard loading fine for this period**

## Screenshots

https://user-images.githubusercontent.com/41326532/179794406-4017eb35-7353-4011-a744-4ca2bda4bab8.mp4

![image](https://user-images.githubusercontent.com/41326532/179792454-526ca755-8694-4951-8e71-2964579a7ce8.png)

## Additional Context

 - Occurred after updating to SK 1.79.0
 - The same occurs if uninstalling and reinstalling the plugin
 - On a site where I could reproduce this error the only network requests on the Site Kit dashboard coming from the plugins highlighted above (in the steps to recreate) are web font related ([example](https://i.imgur.com/q04Y8za.png)). 

## Impacted Users

- https://wordpress.org/support/topic/site-kit-encountered-an-error-67 | **Open** | [SH info](https://docs.google.com/forms/d/1lhjns2K7c0Ny7ME4gxTcsnYtNUtcMr8BvFpPQi23Y1E/edit#response=ACYDBNiRrSW67lJS12uyKia-Yv06b5M6VxSTn_lkfP6A3L9ilI7MTBakUfX6S6cn31hkCZQ)
- https://wordpress.org/support/topic/site-kit-encountered-an-error-68 | **Open** | [SH info](https://docs.google.com/forms/d/1lhjns2K7c0Ny7ME4gxTcsnYtNUtcMr8BvFpPQi23Y1E/edit#response=ACYDBNiKFzVpPwM7a4RZI4xZdZAI9G7pgVm-ii5ftDazw30zFTvDqK5tGQ6cOsnhz38jzzE) 
 - https://wordpress.org/support/topic/cannot-read-properties-of-null-reading-__eventtarget/ | Open | No SH info

## Insights & Troubleshooting Checks performed 

- [ ] Health Check & Troubleshooting plugin check
- [x] Front end check (mixed content, minified content, combined JS)
- [x] Incognito mode 
- [x] Checked users Site Health info, no identified causes
- [x] Uninstalled and reinstalled plugin
- [ ] User confirmed no ""Critical errors"" in their Site Health status
- [ ] Attempted setup/viewing dashboards from another device 
- [ ] Temporary disabled WAF rules at host level - including ModSecurity
- [ ] Checked site setup and dashboard from another WordPress site on the same hosting platform
- [ ] Reset Site Kit
- [ ] Reviewed error logs 
- [ ] Reviewed XHR requests 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* When the Google Web Font loader script is loaded (eg. when the `WebFont` JS global exists), Google Charts components should not throw an exception (causing the entire Dashboard to crash/error).

## Implementation Brief

* Do not use the `fontName` (eg. don't supply a custom `fontName` property) to any `react-google-charts` component when the `WebFont` global exists in JS.
* Merge #5580.

### Test Coverage

* None needed.

## QA Brief

* Follow the steps to reproduce above, or use one of the PHP blocks in the first section of this comment: https://github.com/google/site-kit-wp/issues/5572#issuecomment-1190433825 in a custom plugin/`functions.php` file on a site with Analytics setup (and with Analytics data).
* There should be no error on the dashboard when using the latest `main`/`1.79.1`. To verify the fix, make sure you can replicate the issue on the same site using Site Kit version `1.79.0`.

## Changelog entry

* Fix an issue that could cause Site Kit's dashboard to crash when a theme/plugin loads the Google Web Font Loader JS on Site Kit screens.

**Comment by jamesozzie** [comment]
I've been able to recreate this by installing the [futurio-extra/](https://en-gb.wordpress.org/plugins/futurio-extra/) plugin, when using the Futurio theme with one of their demo installs. Without the theme demo installed via their site library import options the error didn't appear. Further testing ongoing. 

```
index.js?68d0:55 Caught an error: Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (react-dom.development.js?61bb:396:1)
    at invokeGuardedCallback (react-dom.development.js?61bb:440:1)
    at safelyCallDestroy (react-dom.development.js?61bb:21933:1)
    at eval (react-dom.development.js?61bb:22374:1)
    at unstable_runWithPriority (scheduler.development.js?3069:697:1)
    at runWithPriority$2 (react-dom.development.js?61bb:12149:1)
    at commitUnmount (react-dom.development.js?61bb:22367:1)
    at commitNestedUnmounts (react-dom.development.js?61bb:22486:1)
    at unmountHostComponents (react-dom.development.js?61bb:22810:1)
    at commitDeletion (react-dom.development.js?61bb:22896:1) ObjectcomponentStack: "\n    in GoogleChart (created by UserDimensionsPieChart)\n    in div (created by UserDimensionsPieChart)\n    in div (created by UserDimensionsPieChart)\n    in UserDimensionsPieChart (created by DashboardAllTrafficWidget)\n    in div (created by Cell)\n    in Cell (created by DashboardAllTrafficWidget)\n    in div (created by ForwardRef)\n    in ForwardRef (created by DashboardAllTrafficWidget)\n    in div (created by ForwardRef)\n    in ForwardRef (created by DashboardAllTrafficWidget)\n    in div (created by Widget)\n    in div (created by Widget)\n    in Widget (created by WithWidgetSlug(Widget))\n    in WithWidgetSlug(Widget) (created by DashboardAllTrafficWidget)\n    in DashboardAllTrafficWidget (created by WhenAnalyticsActive(DashboardAllTrafficWidget))\n    in WhenAnalyticsActive(DashboardAllTrafficWidget) (created by WidgetRenderer)\n    in WidgetRenderer (created by WidgetAreaRenderer)\n    in div (created by Cell)\n    in Cell (created by WidgetCellWrapper)\n    in WidgetCellWrapper (created by WidgetAreaRenderer)\n    in div (created by ForwardRef)\n    in ForwardRef (created by WidgetAreaRenderer)\n    in div (created by WidgetAreaRenderer)\n    in div (created by ForwardRef)\n    in ForwardRef (created by WidgetAreaRenderer)\n    in WidgetAreaRenderer (created by WidgetContextRenderer)\n    in div (created by WidgetContextRenderer)\n    in WidgetContextRenderer (created by DashboardMainApp)\n    in DashboardMainApp (created by DashboardEntryPoint)\n    in DashboardEntryPoint\n    in RestoreSnapshots (created by Root)\n    in ErrorHandler (created by Root)\n    in StrictMode (created by Root)\n    in Root"[[Prototype]]: Object
componentDidCatch @ index.js?68d0:55
react-dom.development.js?61bb:530 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in Navigation (created by DashboardNavigation)
    in DashboardNavigation (created by Header)
    in Header (created by DashboardMainApp)
    in DashboardMainApp (created by DashboardEntryPoint)
    in DashboardEntryPoint
    in RestoreSnapshots (created by Root)
```

<details><summary>Full Console Errors for this test site</summary>

```
JQMIGRATE: Migrate is installed with logging active, version 3.3.2
/wp-admin/admin.php?page=googlesitekit-dashboard#traffic:1 Error handling response: TypeError: self.processResponse is not a function
    at chrome-extension://cmkdbmfndkfgebldhnkbfhlneefdaaip/js/notification.js:154:10
react-dom.development.js?61bb:530 Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of Transition which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://fb.me/react-strict-mode-find-node
    in div (created by Transition)
    in Transition (created by ForwardRef(Grow))
    in ForwardRef(Grow) (created by ForwardRef(Popper))
    in div (created by ForwardRef(Popper))
    in ForwardRef(Portal) (created by ForwardRef(Popper))
    in ForwardRef(Popper) (created by ForwardRef(Tooltip))
    in ForwardRef(Tooltip) (created by WithStyles(ForwardRef(Tooltip)))
    in WithStyles(ForwardRef(Tooltip)) (created by Button)
    in Button (created by DateRangeSelector)
    in div (created by DateRangeSelector)
    in DateRangeSelector (created by DashboardMainApp)
    in div (created by Cell)
    in Cell (created by Header)
    in div (created by ForwardRef)
    in ForwardRef (created by Header)
    in div (created by ForwardRef)
    in ForwardRef (created by Header)
    in header (created by Header)
    in Header (created by DashboardMainApp)
    in DashboardMainApp (created by DashboardEntryPoint)
    in DashboardEntryPoint
    in RestoreSnapshots (created by Root)
    in ErrorHandler (created by Root)
    in StrictMode (created by Root)
    in Root
warningWithoutStack @ react-dom.development.js?61bb:530
jsapi_compiled_default_module.js:412 Uncaught TypeError: Cannot read properties of null (reading '__eventTarget')
    at gvjs_Zm (jsapi_compiled_default_module.js:412:29)
    at Object.gvjs_Ym [as addListener] (jsapi_compiled_default_module.js:411:392)
    at eval (GoogleChart.js?8bf5:153:1)
    at commitHookEffectList (react-dom.development.js?61bb:22030:1)
    at commitLifeCycles (react-dom.development.js?61bb:22080:1)
    at commitLayoutEffects (react-dom.development.js?61bb:25344:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js?61bb:336:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?61bb:385:1)
    at invokeGuardedCallback (react-dom.development.js?61bb:440:1)
    at commitRootImpl (react-dom.development.js?61bb:25082:1)
2jsapi_compiled_default_module.js:412 Uncaught TypeError: Cannot read properties of null (reading '__eventTarget')
    at gvjs_Zm (jsapi_compiled_default_module.js:412:29)
    at Object.gvjs_4m [as removeAllListeners] (jsapi_compiled_default_module.js:411:711)
    at eval (GoogleChart.js?8bf5:140:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js?61bb:336:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js?61bb:385:1)
    at invokeGuardedCallback (react-dom.development.js?61bb:440:1)
    at safelyCallDestroy (react-dom.development.js?61bb:21933:1)
    at eval (react-dom.development.js?61bb:22374:1)
    at unstable_runWithPriority (scheduler.development.js?3069:697:1)
    at runWithPriority$2 (react-dom.development.js?61bb:12149:1)
react-dom.development.js?61bb:21843 The above error occurred in the <GoogleChart> component:
    in GoogleChart (created by UserDimensionsPieChart)
    in div (created by UserDimensionsPieChart)
    in div (created by UserDimensionsPieChart)
    in UserDimensionsPieChart (created by DashboardAllTrafficWidget)
    in div (created by Cell)
    in Cell (created by DashboardAllTrafficWidget)
    in div (created by ForwardRef)
    in ForwardRef (created by DashboardAllTrafficWidget)
    in div (created by ForwardRef)
    in ForwardRef (created by DashboardAllTrafficWidget)
    in div (created by Widget)
    in div (created by Widget)
    in Widget (created by WithWidgetSlug(Widget))
    in WithWidgetSlug(Widget) (created by DashboardAllTrafficWidget)
    in DashboardAllTrafficWidget (created by WhenAnalyticsActive(DashboardAllTrafficWidget))
    in WhenAnalyticsActive(DashboardAllTrafficWidget) (created by WidgetRenderer)
    in WidgetRenderer (created by WidgetAreaRenderer)
    in div (created by Cell)
    in Cell (created by WidgetCellWrapper)
    in WidgetCellWrapper (created by WidgetAreaRenderer)
    in div (created by ForwardRef)
    in ForwardRef (created by WidgetAreaRenderer)
    in div (created by WidgetAreaRenderer)
    in div (created by ForwardRef)
    in ForwardRef (created by WidgetAreaRenderer)
    in WidgetAreaRenderer (created by WidgetContextRenderer)
    in div (created by WidgetContextRenderer)
    in WidgetContextRenderer (created by DashboardMainApp)
    in DashboardMainApp (created by DashboardEntryPoint)
    in DashboardEntryPoint
    in RestoreSnapshots (created by Root)
    in ErrorHandler (created by Root)
    in StrictMode (created by Root)
    in Root

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorHandler.
logCapturedError @ react-dom.development.js?61bb:21843
index.js?68d0:55 Caught an error: Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (react-dom.development.js?61bb:396:1)
    at invokeGuardedCallback (react-dom.development.js?61bb:440:1)
    at commitRootImpl (react-dom.development.js?61bb:25082:1)
    at unstable_runWithPriority (scheduler.development.js?3069:697:1)
    at runWithPriority$2 (react-dom.development.js?61bb:12149:1)
    at commitRoot (react-dom.development.js?61bb:24922:1)
    at finishSyncRender (react-dom.development.js?61bb:24329:1)
    at performSyncWorkOnRoot (react-dom.development.js?61bb:24307:1)
    at eval (react-dom.development.js?61bb:12199:1)
    at unstable_runWithPriority (scheduler.development.js?3069:697:1) Object
componentDidCatch @ index.js?68d0:55
react-dom.development.js?61bb:21843 The above error occurred in the <GoogleChart> component:
    in GoogleChart (created by UserCountGraph)
    in div (created by UserCountGraph)
    in UserCountGraph (created by DashboardAllTrafficWidget)
    in div (created by Cell)
    in Cell (created by DashboardAllTrafficWidget)
    in div (created by ForwardRef)
    in ForwardRef (created by DashboardAllTrafficWidget)
    in div (created by ForwardRef)
    in ForwardRef (created by DashboardAllTrafficWidget)
    in div (created by Widget)
    in div (created by Widget)
    in Widget (created by WithWidgetSlug(Widget))
    in WithWidgetSlug(Widget) (created by DashboardAllTrafficWidget)
    in DashboardAllTrafficWidget (created by WhenAnalyticsActive(DashboardAllTrafficWidget))
    in WhenAnalyticsActive(DashboardAllTrafficWidget) (created by WidgetRenderer)
    in WidgetRenderer (created by WidgetAreaRenderer)
    in div (created by Cell)
    in Cell (created by WidgetCellWrapper)
    in WidgetCellWrapper (created by WidgetAreaRenderer)
    in div (created by ForwardRef)
    in ForwardRef (created by WidgetAreaRenderer)
    in div (created by WidgetAreaRenderer)
    in div (created by ForwardRef)
    in ForwardRef (created by WidgetAreaRenderer)
    in WidgetAreaRenderer (created by WidgetContextRenderer)
    in div (created by WidgetContextRenderer)
    in WidgetContextRenderer (created by DashboardMainApp)
    in DashboardMainApp (created by DashboardEntryPoint)
    in DashboardEntryPoint
    in RestoreSnapshots (created by Root)
    in ErrorHandler (created by Root)
    in StrictMode (created by Root)
    in Root

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorHandler.
logCapturedError @ react-dom.development.js?61bb:21843
index.js?68d0:55 Caught an error: Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (react-dom.development.js?61bb:396:1)
    at invokeGuardedCallback (react-dom.development.js?61bb:440:1)
    at safelyCallDestroy (react-dom.development.js?61bb:21933:1)
    at eval (react-dom.development.js?61bb:22374:1)
    at unstable_runWithPriority (scheduler.development.js?3069:697:1)
    at runWithPriority$2 (react-dom.development.js?61bb:12149:1)
    at commitUnmount (react-dom.development.js?61bb:22367:1)
    at commitNestedUnmounts (react-dom.development.js?61bb:22486:1)
    at unmountHostComponents (react-dom.development.js?61bb:22810:1)
    at commitDeletion (react-dom.development.js?61bb:22896:1) ObjectcomponentStack: "\n    in GoogleChart (created by UserCountGraph)\n    in div (created by UserCountGraph)\n    in UserCountGraph (created by DashboardAllTrafficWidget)\n    in div (created by Cell)\n    in Cell (created by DashboardAllTrafficWidget)\n    in div (created by ForwardRef)\n    in ForwardRef (created by DashboardAllTrafficWidget)\n    in div (created by ForwardRef)\n    in ForwardRef (created by DashboardAllTrafficWidget)\n    in div (created by Widget)\n    in div (created by Widget)\n    in Widget (created by WithWidgetSlug(Widget))\n    in WithWidgetSlug(Widget) (created by DashboardAllTrafficWidget)\n    in DashboardAllTrafficWidget (created by WhenAnalyticsActive(DashboardAllTrafficWidget))\n    in WhenAnalyticsActive(DashboardAllTrafficWidget) (created by WidgetRenderer)\n    in WidgetRenderer (created by WidgetAreaRenderer)\n    in div (created by Cell)\n    in Cell (created by WidgetCellWrapper)\n    in WidgetCellWrapper (created by WidgetAreaRenderer)\n    in div (created by ForwardRef)\n    in ForwardRef (created by WidgetAreaRenderer)\n    in div (created by WidgetAreaRenderer)\n    in div (created by ForwardRef)\n    in ForwardRef (created by WidgetAreaRenderer)\n    in WidgetAreaRenderer (created by WidgetContextRenderer)\n    in div (created by WidgetContextRenderer)\n    in WidgetContextRenderer (created by DashboardMainApp)\n    in DashboardMainApp (created by DashboardEntryPoint)\n    in DashboardEntryPoint\n    in RestoreSnapshots (created by Root)\n    in ErrorHandler (created by Root)\n    in StrictMode (created by Root)\n    in Root"[[Prototype]]: Objectconstructor: ƒ Object()hasOwnProperty: ƒ hasOwnProperty()isPrototypeOf: ƒ isPrototypeOf()propertyIsEnumerable: ƒ propertyIsEnumerable()toLocaleString: ƒ toLocaleString()toString: ƒ toString()valueOf: ƒ valueOf()__defineGetter__: ƒ __defineGetter__()__defineSetter__: ƒ __defineSetter__()__lookupGetter__: ƒ __lookupGetter__()__lookupSetter__: ƒ __lookupSetter__()__proto__: (...)get __proto__: ƒ __proto__()set __proto__: ƒ __proto__()
componentDidCatch @ index.js?68d0:55
react-dom.development.js?61bb:21843 The above error occurred in the <GoogleChart> component:
    in GoogleChart (created by UserDimensionsPieChart)
    in div (created by UserDimensionsPieChart)
    in div (created by UserDimensionsPieChart)
    in UserDimensionsPieChart (created by DashboardAllTrafficWidget)
    in div (created by Cell)
    in Cell (created by DashboardAllTrafficWidget)
    in div (created by ForwardRef)
    in ForwardRef (created by DashboardAllTrafficWidget)
    in div (created by ForwardRef)
    in ForwardRef (created by DashboardAllTrafficWidget)
    in div (created by Widget)
    in div (created by Widget)
    in Widget (created by WithWidgetSlug(Widget))
    in WithWidgetSlug(Widget) (created by DashboardAllTrafficWidget)
    in DashboardAllTrafficWidget (created by WhenAnalyticsActive(DashboardAllTrafficWidget))
    in WhenAnalyticsActive(DashboardAllTrafficWidget) (created by WidgetRenderer)
    in WidgetRenderer (created by WidgetAreaRenderer)
    in div (created by Cell)
    in Cell (created by WidgetCellWrapper)
    in WidgetCellWrapper (created by WidgetAreaRenderer)
    in div (created by ForwardRef)
    in ForwardRef (created by WidgetAreaRenderer)
    in div (created by WidgetAreaRenderer)
    in div (created by ForwardRef)
    in ForwardRef (created by WidgetAreaRenderer)
    in WidgetAreaRenderer (created by WidgetContextRenderer)
    in div (created by WidgetContextRenderer)
    in WidgetContextRenderer (created by DashboardMainApp)
    in DashboardMainApp (created by DashboardEntryPoint)
    in DashboardEntryPoint
    in RestoreSnapshots (created by Root)
    in ErrorHandler (created by Root)
    in StrictMode (created by Root)
    in Root

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorHandler.
logCapturedError @ react-dom.development.js?61bb:21843
index.js?68d0:55 Caught an error: Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    at Object.invokeGuardedCallbackDev (react-dom.development.js?61bb:396:1)
    at invokeGuardedCallback (react-dom.development.js?61bb:440:1)
    at safelyCallDestroy (react-dom.development.js?61bb:21933:1)
    at eval (react-dom.development.js?61bb:22374:1)
    at unstable_runWithPriority (scheduler.development.js?3069:697:1)
    at runWithPriority$2 (react-dom.development.js?61bb:12149:1)
    at commitUnmount (react-dom.development.js?61bb:22367:1)
    at commitNestedUnmounts (react-dom.development.js?61bb:22486:1)
    at unmountHostComponents (react-dom.development.js?61bb:22810:1)
    at commitDeletion (react-dom.development.js?61bb:22896:1) ObjectcomponentStack: "\n    in GoogleChart (created by UserDimensionsPieChart)\n    in div (created by UserDimensionsPieChart)\n    in div (created by UserDimensionsPieChart)\n    in UserDimensionsPieChart (created by DashboardAllTrafficWidget)\n    in div (created by Cell)\n    in Cell (created by DashboardAllTrafficWidget)\n    in div (created by ForwardRef)\n    in ForwardRef (created by DashboardAllTrafficWidget)\n    in div (created by ForwardRef)\n    in ForwardRef (created by DashboardAllTrafficWidget)\n    in div (created by Widget)\n    in div (created by Widget)\n    in Widget (created by WithWidgetSlug(Widget))\n    in WithWidgetSlug(Widget) (created by DashboardAllTrafficWidget)\n    in DashboardAllTrafficWidget (created by WhenAnalyticsActive(DashboardAllTrafficWidget))\n    in WhenAnalyticsActive(DashboardAllTrafficWidget) (created by WidgetRenderer)\n    in WidgetRenderer (created by WidgetAreaRenderer)\n    in div (created by Cell)\n    in Cell (created by WidgetCellWrapper)\n    in WidgetCellWrapper (created by WidgetAreaRenderer)\n    in div (created by ForwardRef)\n    in ForwardRef (created by WidgetAreaRenderer)\n    in div (created by WidgetAreaRenderer)\n    in div (created by ForwardRef)\n    in ForwardRef (created by WidgetAreaRenderer)\n    in WidgetAreaRenderer (created by WidgetContextRenderer)\n    in div (created by WidgetContextRenderer)\n    in WidgetContextRenderer (created by DashboardMainApp)\n    in DashboardMainApp (created by DashboardEntryPoint)\n    in DashboardEntryPoint\n    in RestoreSnapshots (created by Root)\n    in ErrorHandler (created by Root)\n    in StrictMode (created by Root)\n    in Root"[[Prototype]]: Object
componentDidCatch @ index.js?68d0:55
react-dom.development.js?61bb:530 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in Navigation (created by DashboardNavigation)
    in DashboardNavigation (created by Header)
    in Header (created by DashboardMainApp)
    in DashboardMainApp (created by DashboardEntryPoint)
    in DashboardEntryPoint
    in RestoreSnapshots (created by Root)
```

</details>

The above is from an active support test site.

**Comment by jamesozzie** [comment]
Just a quick comment that I've added steps to reproduce this within the main content of this issue. In my case a quick refresh of the Site Kit dashboard results in this error no longer occurring.

**Comment by jimmymadon** [comment]
@jamesozzie Thanks for all that information. I can reproduce this issue locally and on an InstaWP site. Locally, it happens even when selecting the 7 days period - although on the InstaWP site it only occurred for 14 or 90 day periods. When I comment out the [latest additions to GoogleChart](https://github.com/google/site-kit-wp/blob/24e6a6368ce49d8eab433a052d1006c5aabe1e7a/assets/js/components/GoogleChart.js#L228-L250), I do not get the error so this has been introduced in the last release. Not sure why those lines are an issue - haven't dug deeper. Also not sure why this happens only with the install of Futurio either at this stage. But worth moving this to the Execution board as a bug to investigate further I guess?

I couldn't reproduce the Data Errors - Invalid Credentials errors.

**Comment by eugene-manuilov** [comment]
Looks like this is a compatibility issue with the Web Font Loader library.

The following is a snippet that can help reproduce the issue on your local machine even without futurio plugin and theme:

```php
add_action( 'admin_footer', function() {
	?><script src="https://ajax.googleapis.com/ajax/libs/webfont/1.6.26/webfont.js"></script>
	<script>
	  WebFont.load({
		google: {
		  families: ['Droid Sans', 'Droid Serif']
		}
	  });
	</script><?php
} );
```

**Comment by jamesozzie** [comment]
[One user](https://wordpress.org/support/topic/site-kit-encountered-an-error-67/) reports that the error doesn't exist when switching to a Chrome browser incognito window. Only during a regular Chrome browser session does this occur for this user.

**Comment by eugene-manuilov** [comment]
@tofumatt could you please help to troubleshoot this fonts issue? Assigning this ticket to you for now.

**Comment by jamesozzie** [comment]
Some notes from my testing. The error output in Firefox differs slightly - `a is null`, although the same issue occurs, the dashboard breaks. Recording of cross browser testing, highlighting this slightly different error below. 

https://user-images.githubusercontent.com/41326532/179999766-65118be4-fa50-4c87-b02b-7043410f0e69.mp4

Separately, back in Chrome, I only encounter this with the Analytics module active. As soon as I disconnect Analytics, I can no longer recreate the issue. With a separate administrator account, signed in to Site Kit using a Google account which doesn't have access to the Analytics account, the error doesn't occur. 

Could there be something within the Analytics codebase contributing to this?

**Comment by tofumatt** [comment]
It's not Analytics but the React Google Charts library, or our usage of it. Enabling Analytics renders the Google Charts `PieChart` with `fontName: 'Google Sans Text'` which is what's causing the error.

I'm still trying to figure out a cause, but as mentioned it might be best to disable custom fonts in our charts for now…

**Comment by tofumatt** [comment]
So, this all seems to stem from the `WebFont` global being written/available when React Google Charts tries to load a custom font for certain elements like the text in a pie chart slice. I think it only happens for content it renders inside an SVG, where we need to specify a custom font rather than it just relying on the plugin-wide fonts set here: https://github.com/google/site-kit-wp/blob/24e6a6368ce49d8eab433a052d1006c5aabe1e7a/assets/sass/config/_variables.scss#L242-L243

This can be seen with Eugene's code above; even just using:

```php
add_action( 'admin_footer', function() {
    ?>
    <script src="https://ajax.googleapis.com/ajax/libs/webfont/1.6.26/webfont.js"></script>
    <?php
} );
```

will cause the bug to occur. In fact, even:

```php
add_action( 'admin_footer', function() {
    ?>
    <script>window.WebFont = { load: console.log }</script>
    <?php
} );
```

causes the bug.

------

This only happens when another plugin/theme/etc. loads the Google WebFont loader, possible another version, and only affects a few, small elements rendered by our Charts. I am totally unaware of the _actual_ root cause because the errors are opaque and all of this code is third-party and a bit of a black box. But I have found a reliable fix that only needs to be used when the `WebFont` global is loaded and it causes minimal issue.

I think the best fix is to load those web fonts in Google Chart components as such:

```js
// Instead of this:
textStyle: {
	fontName: '"Google Sans Text", "Helvetica Neue", Helvetica, Arial, sans-serif',
	color: '#5f6561',
	fontSize: 10,
}

// Use this:
textStyle: {
	fontName:
		global?.WebFont?.load === undefined
			? '"Google Sans Text", "Helvetica Neue", Helvetica, Arial, sans-serif'
			: undefined,
	color: '#5f6561',
	fontSize: 10,
}
```

It fixes the problem and only causes a few, small items to be rendered using Arial instead of Google Sans. Here's the different for the UserDimensionsPieChart.

### Google Sans

<img width="1523" alt="CleanShot 2022-07-20 at 16 30 11" src="https://user-images.githubusercontent.com/90871/180022810-7dc8464c-f8f7-42e1-9e4c-d9b181be0d7e.png">

### Arial

<img width="1523" alt="CleanShot 2022-07-20 at 16 31 00" src="https://user-images.githubusercontent.com/90871/180022851-5c0cf987-3f84-472b-b3a0-fa6dfc0430c3.png">

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by wpdarren** [comment]
QA Update: ❌ 

As per conversation with Evan, the error is appearing when on main branch, and following the steps to reproduce and the QAB.

**Comment by aaemnnosttv** [comment]
## QA ❌ 

@tofumatt while testing, we found that it is still possible to raise the same error while toggling the date range. I wasn't able to reproduce this at first but was able to eventually so this does not seem to happen consistently. Perhaps there is a bit of a race condition going on but this warrants a bit more testing and exploration.

**Comment by aaemnnosttv** [comment]
@tofumatt After digging into this a bit more with a dev build, I found it is related to these lines where we add and remove listeners where `.getChart()` is returning `null`.

https://github.com/google/site-kit-wp/blob/d9fb05950cd55d4a3e11c8f78b38a6a1af149963/assets/js/components/GoogleChart.js#L154

https://github.com/google/site-kit-wp/blob/d9fb05950cd55d4a3e11c8f78b38a6a1af149963/assets/js/components/GoogleChart.js#L140

**Comment by tofumatt** [comment]
I saw those lines in trace errors when in development mode, but commenting them out didn't seem to help, so I thought it was a red herring. I'll investigate further though, thanks 👍🏻

**Comment by tofumatt** [comment]
I'm able to get intermittent errors as mentioned by @wpdarren. Oddly, if I conditionally run the code that's causing the errors (code Evan pointed out, plus a few other lines), there are no errors, but sometimes the charts just don't load.

The most consistent fix seems to be omitting the `fontName` property _all of the time, regardless of `WebFont` global_.

It seems this might just be a bug with either Google React Charts or with Google Charts.

Given how subtle the usage is, I think the best workaround fix for now is to omit the Google Sans font on small sections of our charts (eg. anywhere we are supplying `fontName`).

A more robust fix for the future could be explored, but for now the pie chart percentage label in Arial seems better than a crashing dashboard 🙂 

I've made a PR that removes the `fontName` prop and removes my utility function added in #5580.

See:

**Comment by aaemnnosttv** [comment]
Thanks for the follow up @tofumatt, this has been merged 👍 

@wpdarren this should be ready for you to test again on `main` whenever you're ready!

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- On a site where the issue could be recreated following the steps in the ticket:  When I switched to the main branch, the error disappeared.  I then switched between the date ranges, and the error does not reappear.  I tested this on main and entity and shared dashboard to ensure the behaviour was consistent. 
- Since the issue appeared to be with the All traffic widget pie chart, I tested to make sure that this widget functioned as expected. I also browsed through the dashboard to ensure no other widgets triggered this error message. 

https://user-images.githubusercontent.com/73545194/180157055-8794a651-32c5-4c30-ab5c-6986ec4dbdc6.mp4

**Comment by alekb** [comment]
Is it possible to downgrade to a prior version until this issue is resolved?

**Comment by bethanylang** [comment]
@alekb The fix for this was just released in v1.79.1. :) You can update the plugin and the issue should be resolved now!
